pkg/integration/tests/branch/checkout_by_name.go

```diff
@@ -0,0 +1,38 @@
+package branch
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var CheckoutByName = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Try to checkout branch by name.",
+	ExtraCmdArgs: "",
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.
+			CreateNCommits(3).
+			NewBranch("@").
```

Comment:
what's special about a branch named '@'? Are there other branch names that can trigger the bad behaviour?

Reply:
makes sense!